api/route/route.go

```diff
@@ -34,11 +34,100 @@ func AppendRoute(r *gin.Engine) *gin.Engine {
 	r.GET("/apisix/admin/routes/:rid", findRoute)
 	r.GET("/apisix/admin/routes", listRoute)
 	r.PUT("/apisix/admin/routes/:rid", updateRoute)
+	r.PUT("/apisix/admin/routes/:rid/publish", publishRoute)
 	r.DELETE("/apisix/admin/routes/:rid", deleteRoute)
 	r.GET("/apisix/admin/notexist/routes", isRouteExist)
+	r.PUT("/apisix/admin/routes/:rid/offline", offlineRoute)
 	return r
 }
 
+func publishRoute(c *gin.Context) {
+	rid := c.Param("rid")
+	r := &service.Route{}
+	tx := conf.DB().Begin()
+	if err := tx.Model(&service.Route{}).Where("id = ?", rid).Update("status", true).Find(&r).Error; err != nil {
+		tx.Rollback()
+		e := errno.FromMessage(errno.RoutePublishError, err.Error())
+		logger.Error(e.Msg)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, e.Response())
+		return
+	} else {
+		routeRequest := &service.RouteRequest{}
+		if err := json.Unmarshal([]byte(r.Content), routeRequest); err != nil {
+			tx.Rollback()
+			e := errno.FromMessage(errno.RoutePublishError, err.Error())
+			logger.Error(e.Msg)
+			c.AbortWithStatusJSON(http.StatusInternalServerError, e.Response())
+			return
+		}
+		arr := service.ToApisixRequest(routeRequest)
+		var resp *service.ApisixRouteResponse
+		if resp, err = arr.Create(rid); err != nil {
```

Comment:
a little confused about `Create`, it should be `Update` or something here.

Reply:
in this pr, we divided the process of creating a route into two steps：
create a route （without publish）only in mysql, and do `publish` to create a route in apisix.

so here we use `create`